packages/nouns-webapp/src/components/VoteCard/index.tsx

```diff
@@ -19,54 +22,106 @@ interface VoteCardProps {
   percentage: number;
   nounIds: Array<string>;
   variant: VoteCardVariant;
+  delegateView: boolean;
+  delegateGroupedVoteData:
+    | { delegate: string; supportDetailed: 0 | 1 | 2; nounsRepresented: string[] }[]
+    | undefined;
 }
 
 const VoteCard: React.FC<VoteCardProps> = props => {
-  const { proposal, percentage, nounIds, variant } = props;
-  const isMobile = isMobileScreen();
+  const { proposal, percentage, nounIds, variant, delegateView, delegateGroupedVoteData } = props;
 
   let titleClass;
   let titleCopy;
   let voteCount;
+  let supportDetailedValue: 0 | 1 | 2;
   switch (variant) {
     case VoteCardVariant.FOR:
       titleClass = classes.for;
       titleCopy = <Trans>For</Trans>;
       voteCount = proposal.forCount;
+      supportDetailedValue = 1;
       break;
     case VoteCardVariant.AGAINST:
       titleClass = classes.against;
       titleCopy = <Trans>Against</Trans>;
       voteCount = proposal.againstCount;
+      supportDetailedValue = 0;
       break;
     default:
       titleClass = classes.abstain;
       titleCopy = <Trans>Abstain</Trans>;
       voteCount = proposal.abstainCount;
+      supportDetailedValue = 2;
       break;
   }
 
+  const { library } = useEthers();
+  const [ensCached, setEnsCached] = useState(false);
+
+  // Pre-fetch ENS  of delegates (with 30min TTL)
+  // This makes hover cards load more smoothly
+  useEffect(() => {
+    if (!delegateGroupedVoteData || !library || ensCached) {
+      return;
+    }
+
+    delegateGroupedVoteData.forEach((delegateInfo: { delegate: string }) => {
+      if (localStorage.getItem(`${delegateInfo.delegate}`)) {
+        return;
+      }
+
+      library
+        .lookupAddress(delegateInfo.delegate)
+        .then(name => {
+          // Store data as mapping of address_Expiration => address or ENS
+          if (name) {
+            localStorage.setItem(
+              `${delegateInfo.delegate}`,
```

Comment:
Can you add a [cache bucket and use cacheKey](https://github.com/nounsDAO/nouns-monorepo/blob/master/packages/nouns-webapp/src/config.ts#L27-L36) for this, rather than using the address as the top level key in an effort to keep local storage structured?